### ISSUE TITLE
LFQA: FiD model for the ELI5 dataset

### ISSUE THREAD

**Issue by mihaela-bornea** [primeqa/primeqa#300]
Provide support for long form QA with run_mrc.py. Implement FiD which is a generative, seq2seq model for QA.
Support ELI5 dataset.

Components:
- Preprocessor
- FiD model 
- Data collator
- Postprocessor
- Metric: Rouge

**Comment by mihaela-bornea** [comment]
All components have been implemented and integrated in run_mrc.py. Currently debugging performance drop (1 point) on ELI5.

**Comment by jdpsen** [comment]
TBD: ELI5 style Inpt handling

**Comment by mihaela-bornea** [comment]
Rebased the branch and tested the run_mrc.py with the FID model for ELI5. 

Created a draft PR.

Additional tasks:
- [ ] Unit Tests
- [ ] tydiqa test run
- [ ] test with T5
- [ ] Decide on the input format and integration with IR

**Comment by jdpsen** [comment]
Compatible with BART/T5 not roberta , needs encoder-decoder base.
ToDo: Unit tests, documentation on the implicit assumptions.

**Comment by jdpsen** [comment]
FiD with T5 implementation done. Testing in progress.

**Comment by jdpsen** [comment]
@mihaela-bornea : can you please provide the updates you mentioned last day. W can put all the integration related trial/errors here (barring any new research idea/experiments)

**Comment by mihaela-bornea** [comment]
T5 is supported in FID but the performance on the ELI5 task is not on par with BART. We decided to document BART based FID only at this point.

The unit tests are now completed. 

FiD is now on hold until the IR - MRC integration is done.

**Comment by mihaela-bornea** [comment]
Working on a pipeline to integrate retrieval and reader (MRC).

**Comment by jdpsen** [comment]
unit tests done.
script to run the pipleine in decode mode with IR and MRC , in /primeqa/orqa.

**Comment by jdpsen** [comment]
Update from Mihaela
"The pipelines branch is merged in the main and I am now continuing with the FiD related Reader and QAPipeline"

**Comment by jdpsen** [comment]
PR ready for review: https://github.com/primeqa/primeqa/pull/310

**Comment by mihaela-bornea** [comment]
I worked with Sara and resolved most of her comments.
I am loading the Colbert Checkpoint and the reader model on the HF hub: 
 * update model cards
 * document and test the use of Colbert checkpoint

**Comment by mihaela-bornea** [comment]
The #310  PR was merged
